### PR TITLE
describe CASCADE option

### LINKED ISSUE #1116: Document CASCADE for DROP commands
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/11250

### Which part(s) of the docs might be affected or should be updated? And how?

Document CASCADE for DROP commands, which drops the specified item and all its dependencies. 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

 Document the CASCADE option in DROP TABLE/SOURCE/SINK/VIEW/MATERIALIZED VIEW/INDEX

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #1116

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/commands/sql-drop-database.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Before you can remove a database, you must use [DROP SCHEMA](sql-drop-schema.md)
 ## Syntax
 
 ```sql
-DROP DATABASE [ IF EXISTS ] database_name;
+DROP DATABASE [ IF EXISTS ] database_name [ CASCADE ];
```
Comment (yezizp2012) [inline docs/sql/commands/sql-drop-database.md]: `CASCADE` is not a option for `drop database`. All relations like tables, views etc and other objects will be dropped automatically with the database when dropped.

File: docs/sql/commands/sql-drop-schema.md [hunk 2]
```diff
@@ -9,14 +9,12 @@ Use the `DROP SCHEMA` command to remove a schema from a database.
 
 Before you can remove a schema, you must remove all its dependent objects (tables, materialized views, etc.).
 
-
 ## Syntax
 
 ```sql
-DROP SCHEMA [ IF EXISTS ] [database_name.]schema_name;
+DROP SCHEMA [ IF EXISTS ] [database_name.]schema_name [ CASCADE ];
```
Comment (yezizp2012) [inline docs/sql/commands/sql-drop-schema.md]: `CASCADE` is not supported yet for schema.

**Review by yezizp2012 — COMMENTED** [review]
Rest LGTM

File: docs/sql/commands/sql-drop-database.md [hunk 3]
```diff
@@ -26,22 +26,20 @@ export const svgtwo = rr.Diagram(
...
 ## Parameters
 
-
 |Parameter or clause        | Description           |
 |---------------------------|-----------------------|
 |**IF EXISTS** clause       |Do not return an error if the specified database does not exist.|
 |*database_name*                 |The name of the database you want to remove. You can use [`SHOW DATABASES`](sql-show-databases.md) to get a list of all available databases.|
-
-
+|`CASCADE` option| If this option is specified, all tables, views, materialized views, functions, and other objects within the database will be dropped along with the database itself. |
```
Comment (kwannoel) [inline docs/sql/commands/sql-drop-database.md]:
```suggestion
```

As per @yezizp2012 's comments.

File: docs/sql/commands/sql-drop-database.md [hunk 4]
```diff
@@ -26,22 +26,20 @@ export const svgtwo = rr.Diagram(
         rr.Terminal('DROP DATABASE'),
         rr.Optional(rr.Terminal('IF EXISTS')),
         rr.NonTerminal('database_name', 'skip'),
+        rr.Optional(rr.Terminal('CASCADE'), 'skip'),
```
Comment (kwannoel) [inline docs/sql/commands/sql-drop-database.md]:
```suggestion
        rr.Optional(rr.Terminal('CASCADE'), 'skip'),
```

**Review by chenzl25 — APPROVED** [review]
LGTM

File: docs/sql/commands/sql-drop-index.md [hunk 5]
```diff
@@ -29,23 +28,23 @@ export const svg = rr.Diagram(
...
 ## Parameters
 
 |Parameter                  | Description           |
 |---------------------------|-----------------------|
 |**IF EXISTS** clause       |Do not return an error if the specified index does not exist.|
 |*schema_name*                   |The schema of the index that you want to remove. <br /> You can use [`SHOW SCHEMAS`](sql-show-schemas.md) to get a list of all available schemas. If you don't specify a schema, the specified index in the default schema `public` will be removed.|
 |*index_name*                    |The name of the index to remove. <br/> You can use [`DESCRIBE`](sql-describe.md) to show the indexes of a table.|
-
-
+|`CASCADE` option| If this option is specified, all objects that depend on the index, and in turn all objects that depend on those objects will be dropped.|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-index.md]:
```suggestion
|**CASCADE** option| If this option is specified, all objects that depend on the index, and in turn all objects that depend on those objects will be dropped.|
```

File: docs/sql/commands/sql-drop-mv.md [hunk 6]
```diff
@@ -32,23 +31,22 @@ export const svg = rr.Diagram(
...
-
 ## Parameters
 
 |Parameter                  | Description           |
 |---------------------------|-----------------------|
 |*schema_name*                   |Specify the name of a schema to remove the materialized view in that schema. You can use [`SHOW SCHEMAS`](sql-show-schemas.md) to get a list of all available schemas. If you don't specify a schema, the specified materialized view in the default schema `public` will be removed.|
 |*mv_name*                       |The name of the materialized view to remove. You can use [`SHOW MATERIALIZED VIEWS`](sql-show-mv.md) to get a list of all available materialized views.|
-
-
+|`CASCADE` option| If this option is specified, all objects (such as other materialized views or regular views) that depend on the materialized view, and in turn all objects that depend on those objects will be dropped.|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-mv.md]:
```suggestion
|**CASCADE** option| If this option is specified, all objects (such as other materialized views or regular views) that depend on the materialized view, and in turn all objects that depend on those objects will be dropped.|
```

File: docs/sql/commands/sql-drop-sink.md [hunk 7]
```diff
@@ -29,22 +28,22 @@ export const svg = rr.Diagram(
...
-
 ## Parameters
 
 |Parameter                  | Description           |
 |---------------------------|-----------------------|
 |*schema_name*                   |The schema of the sink that you want to remove. You can use [`SHOW SCHEMAS`](sql-show-schemas.md) to get a list of all available schemas. If you don't specify a schema, the specified sink in the default schema `public` will be removed.|
 |*sink_name*                   |The name of the sink to remove.|
-
-
+|`CASCADE` option| If this option is specified, all objects (such as materialized views) that depend on the sink, and in turn all objects that depend on those objects will be dropped.|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-sink.md]:
```suggestion
|**CASCADE** option| If this option is specified, all objects (such as materialized views) that depend on the sink, and in turn all objects that depend on those objects will be dropped.|
```

File: docs/sql/commands/sql-drop-source.md [hunk 8]
```diff
@@ -31,22 +30,22 @@ export const svg = rr.Diagram(
...
-
 ## Parameters
 
 |Parameter                  | Description           |
 |---------------------------|-----------------------|
 |*schema_name*                   |The schema of the source that you want to remove. You can use [`SHOW SCHEMAS`](sql-show-schemas.md) to get a list of all available schemas. If you don't specify a schema, the specified source in the default schema `public` will be removed.|
 |*source_name*                   |The name of the source to remove.|
-
-
+|`CASCADE` option| If this option is specified, all objects (such as materialized views) that depend on the source, and in turn all objects that depend on those objects will be dropped.|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-source.md]:
```suggestion
|**CASCADE** option| If this option is specified, all objects (such as materialized views) that depend on the source, and in turn all objects that depend on those objects will be dropped.|
```

File: docs/sql/commands/sql-drop-table.md [hunk 9]
```diff
@@ -31,23 +30,22 @@ export const svg = rr.Diagram(
...
-
 ## Parameters
 
 |Parameter                  | Description           |
 |---------------------------|-----------------------|
 |*schema*                   |Specify the name of a schema to remove the table in that schema. You can use [`SHOW SCHEMAS`](sql-show-schemas.md) to get a list of all available schemas. If you don't specify a schema, the specified source in the default schema `public` will be removed.|
 |*table*                    |The name of the table to remove. You can use [`SHOW TABLES`](sql-show-tables.md) to get a list of all available tables.|
-
-
+|`CASCADE` option| If this option is specified, all objects (such as materialized views) that depend on the table, and in turn all objects that depend on those objects will be dropped.|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-table.md]:
```suggestion
|**CASCADE** option| If this option is specified, all objects (such as materialized views) that depend on the table, and in turn all objects that depend on those objects will be dropped.|
```

File: docs/sql/commands/sql-drop-view.md [hunk 10]
```diff
@@ -10,36 +10,30 @@ Use the `DROP VIEW` command to remove an existing view from a particular schema.
...
 ## Parameters
 
 |Parameter                  | Description           |
 |---------------------------|-----------------------|
 |**IF EXISTS** clause       |Do not return an error if the specified view does not exist.|
 |*view_name*                |Name of the view to be dropped.|
-
-
-
+|`CASCADE` option| If this option is specified, all objects (such as materialized views or other regular views) that depend on the view, and in turn all objects that depend on those objects will be dropped.|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-view.md]:
```suggestion
|**CASCADE** option| If this option is specified, all objects (such as materialized views or other regular views) that depend on the view, and in turn all objects that depend on those objects will be dropped.|
```

**Review by CharlieSYH — APPROVED** [review]

File: docs/sql/commands/sql-drop-index.md [hunk 11]
```diff
@@ -29,23 +28,23 @@ export const svg = rr.Diagram(
             ),
         ),
         rr.NonTerminal('index_name'),
+        rr.Optional(
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-drop-index.md:31]: image

image

I didn't know that before : 0

**Review by yezizp2012 — APPROVED** [review]

**Pull request merged**